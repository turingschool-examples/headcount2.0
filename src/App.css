*, *:before, *:after {
  box-sizing: border-box;
}

.app {
  text-align: center;
  color: cornflowerblue;
}


