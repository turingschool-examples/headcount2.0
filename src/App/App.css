.app {
  width: 100%;
  height: 100%;
  background-color: rgba(37, 72, 105, .8)
}
