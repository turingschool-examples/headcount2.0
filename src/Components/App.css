html {
  background-color: #2a393a;
  color: #f4fcfc;
  font-size: 62.5%;
}

h1 {
  color: #f7b488;
  font-size: 3.5em;
  margin: 0 0 1em 0;
  padding: .5em;
  text-align: center;
}