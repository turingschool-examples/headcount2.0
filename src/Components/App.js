import React, { Component } from "react";
import DistrictRepository from "../helper";
import kindergarners from "../data/kindergartners_in_full_day_program";
// import PropTypes from "prop-types";
import Search from "./Search";
import { DistrictCardContainer } from "./DistrictCardContainer";
import { ControlCards } from "./ControlCards";

class App extends Component {
  constructor() {
    super();
    this.state = {
      districts: [],
      selectedDistricts: [],
      comparisonData: {},
      cardClick: {}
    };
    this.districtRepository = new DistrictRepository(kindergarners);
  }

  componentDidUpdate() {
    if (this.state.selectedDistricts.length === 2) {
      this.compareDistricts(
        this.state.selectedDistricts[0],
        this.state.selectedDistricts[1]
      );
      this.setState({ selectedDistricts: [] });
    }
  }

  componentDidMount = () => {
    const districts = this.districtRepository.districtsArray();
    this.setState({ districts });
  };

  handleSubmit = district => {
    this.setState({
      districts: district
    });
  };

  selectCard = location => {
    if (!this.state.cardClick[location]) {
      const cardClick = { ...this.state.cardClick };
      cardClick[location] = true;
      this.setState({ cardClick });
    } else {
      delete this.state.cardClick[location];
      console.log(this.state.cardClick);
    }
    if (this.state.selectedDistricts.length <= 1) {
      const selectedDistricts = [...this.state.selectedDistricts, location];
      this.setState({ selectedDistricts });
    }
  };

  compareDistricts = (districtOne, districtTwo) => {
    const districtOneInfo = this.districtRepository.findByName(districtOne);
    const districtTwoInfo = this.districtRepository.findByName(districtTwo);
    const districtComparison = this.districtRepository.compareDistrictAverages(
      districtOne,
      districtTwo
    );
    const comparisonData = Object.assign({}, { ...districtComparison });
    comparisonData["comparisonData"] = {
      [districtOne]: comparisonData[districtOne],
      [districtTwo]: comparisonData[districtTwo],
      compared: comparisonData.compared
    };
    delete comparisonData.compared;
    comparisonData[districtOne] = districtOneInfo;
    comparisonData[districtOne].cardClick = null;
    comparisonData[districtTwo] = districtTwoInfo;
    comparisonData[districtTwo].cardClick = null;

    this.setState({ comparisonData });
  };

  render() {
    return (
      <div>
        Welcome To Headcount 2.0
        <Search handleSubmit={this.handleSubmit} />
        {this.state.comparisonData !== {} && (
          <ControlCards
            comparisonData={this.state.comparisonData}
            cardClick={this.state.cardClick}
          />
        )}
        {this.state.districts && (
          <DistrictCardContainer
            districts={this.state.districts}
            selectCard={this.selectCard}
          />
        )}
      </div>
    );
  }
}

export default App;
