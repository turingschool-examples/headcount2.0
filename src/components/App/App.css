/*APP COMPONENT RELATED CSS GOES HERE*/
.App {
  height: 100%;
  width: 100vw;
  margin: 0 auto;
  padding: 0 0 15px 0;
  display: grid;
  grid-template-rows: 25% 25% 1fr;
  grid-template-columns: 1fr;
}