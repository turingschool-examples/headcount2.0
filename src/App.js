import React, { Component } from 'react';
import './App.css';
import Card from './Card'
import DistrictRepository from './helper'
import CardList from './CardList'
import kinderData from '../data/kindergartners_in_full_day_program';

class App extends Component {
  constructor () {
    super()
    this.helper = new DistrictRepository(kinderData)
    this.state = {
      selectedCards: []
    }
  }

  // componentWillMount() {
  //   console.log(kinderData);
  // }

  render() {
    return (
      <div>
        <CardList selectedCards={this.state.selectedCards}
                  helper={this.helper}/>

      </div>
    );
  }
}

export default App;
