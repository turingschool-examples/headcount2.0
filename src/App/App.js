import React, { Component } from 'react';
import './App.css';
import CardContainer from '../CardContainer/CardContainer.js'

class App extends Component {
  constructor( props ){
    super( props );
    this.state = {
      districtsData: {}
    }
  }

  render() {
    return (
      <div> 
        <CardContainer 
          districtsData ={ this.state.districtsData }
        />
      </div>
    );
  }
}


export default App;
