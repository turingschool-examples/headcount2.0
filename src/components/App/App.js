import React, { Component } from 'react';
import CardContainer from '../CardContainer/CardContainer';
import Search from '../Search/Search';
import ComparedDistricts from '../ComparedDistricts/ComparedDistricts';

import DistrictRepository from '../../helper';
import kinderGardenerData from '../../data/kindergartners_in_full_day_program';

import './App.css';

const schoolDistricts = new DistrictRepository(kinderGardenerData);

class App extends Component {
  constructor() {
    super();
    this.state = {
      districts: [],
      comparedDistricts: [
        {
          location: 'hellow',
          stats: {2000: 0.29293}
        }
      ]
    };
  }

  componentDidMount = () => {
    this.getSchoolDistrictData();
  }

  getSchoolDistrictData = () => {
    const districts = Object.values(schoolDistricts.stats);
    this.setState({ districts });
    return districts;
  }

  filterDistricts = (query) => {
    const queriedDistrict = schoolDistricts.findAllMatches(query);

    if (query === '') {
      this.setState({districts: this.getSchoolDistrictData() });
    } else {
      this.setState({districts: queriedDistrict});
    }
  }

  render() {
    const { districts, comparedDistricts } = this.state;

    return (
      <main>
        <ComparedDistricts 
          comparedDistricts={comparedDistricts} 
        />
        <Search
          filterDistricts={this.filterDistricts}
        />
        <CardContainer
          districts={districts}
        />
      </main>
    );
  }
}

export default App;