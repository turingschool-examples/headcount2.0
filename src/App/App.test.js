import React from 'react';
import { shallow } from 'enzyme';

import App from '../App/App';

describe('App test', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should match the snapshot when a App is rendered', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state of an empty array', () => {
    expect(wrapper.state().data).toEqual([]);
  });

  it('should populate school data in state when handleSearch is invoked', () => {
    wrapper.instance().handleUpdate();
    expect(wrapper.state().data.length).toEqual(181);
  });

  it('should populate only the school data that matches the user input', () => {
    const mockUserInput = 'COLORADO';
    wrapper.instance().handleUpdate(mockUserInput);
    expect(wrapper.state().data.length).toEqual(2);
    const mockSearchData = [{
      "location": "COLORADO",
      "stats": {
        "2004": 0.24,
        "2005": 0.278,
        "2006": 0.337,
        "2007": 0.395,
        "2008": 0.536,
        "2009": 0.598,
        "2010": 0.64,
        "2011": 0.672,
        "2012": 0.695,
        "2013": 0.703,
        "2014": 0.741
      }
    }, {
      "location": "COLORADO SPRINGS 11",
      "stats": {
        "2004": 0.069,
        "2005": 0.509,
        "2006": 0.638,
        "2007": 0.994,
        "2008": 0.992,
        "2009": 1,
        "2010": 0.993,
        "2011": 0.994,
        "2012": 0.993,
        "2013": 0.989,
        "2014": 0.994
      }
    }];
    expect(wrapper.state().data).toEqual(mockSearchData);
  });
});
