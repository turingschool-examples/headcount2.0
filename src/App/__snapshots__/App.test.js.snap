// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test should match the snapshot when a App is rendered 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="main-heading"
>
          HeadCount 2.0
</h1>,
        <input
          className="search-bar"
          onChange={[Function]}
          placeholder="Search"
/>,
        <Unknown
          schoolData={Array []}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "HeadCount 2.0",
          "className": "main-heading",
        },
        "ref": null,
        "rendered": "HeadCount 2.0",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search-bar",
          "onChange": [Function],
          "placeholder": "Search",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "schoolData": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="main-heading"
>
            HeadCount 2.0
</h1>,
          <input
            className="search-bar"
            onChange={[Function]}
            placeholder="Search"
/>,
          <Unknown
            schoolData={Array []}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "HeadCount 2.0",
            "className": "main-heading",
          },
          "ref": null,
          "rendered": "HeadCount 2.0",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-bar",
            "onChange": [Function],
            "placeholder": "Search",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "schoolData": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
